bin/varnishd/cache/cache_obj.c

```diff
@@ -253,13 +271,18 @@ ObjWaitExtend(const struct worker *wrk, const struct objcore *oc, uint64_t l)
 	CHECK_OBJ_NOTNULL(oc->boc, BOC_MAGIC);
 	Lck_Lock(&oc->boc->mtx);
 	while (1) {
-		rv = oc->boc->len_so_far;
+		rv = oc->boc->fetched_so_far;
 		assert(l <= rv || oc->boc->state == BOS_FAILED);
+		if (oc->flags & OC_F_PRIVATE) {
+			/* Update with the new client position */
+			oc->boc->delivered_so_far = l;
+			AZ(pthread_cond_broadcast(&oc->boc->cond));
```

Comment:
can there be more than one waiter? Otherwise, should this be `pthread_cond_signal` for clarity?

Reply:
I guess that's fair! Fixed.